lib/index.js

```diff
@@ -717,7 +718,12 @@ exports.merge = (schemas, options = {}) => {
 		if (mergedSchemas.anyOf && mergedSchemas.anyOf.length === 1) {
 			const [ anyOf ] = mergedSchemas.anyOf
 
-			return merge(omit(mergedSchemas, 'anyOf'), anyOf)
+			// Deduplicate required array values
+			const merged = merge(omit(mergedSchemas, 'anyOf'), anyOf)
+			if (isArray(merged.required) && merged.required.length > 1) {
+				merged.required = uniq(merged.required)
+			}
+			return merged
```

Comment:
Probably need to iterate through the entire schema and deduplicate all `required` arrays.